kafka/consumer.go

```diff
@@ -0,0 +1,401 @@
+package kafka
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	confluent "github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/grafana/metrictank/stats"
+	"github.com/raintank/worldping-api/pkg/log"
+	"github.com/twinj/uuid"
+)
+
+var LogLevel int
+
+type Consumer struct {
+	conf             ConsumerConf
+	wg               sync.WaitGroup
+	consumer         *confluent.Consumer
+	Partitions       []int32
+	currentOffsets   map[int32]*int64
+	bootTimeOffsets  map[int32]int64
+	partitionOffset  map[int32]*stats.Gauge64
+	partitionLogSize map[int32]*stats.Gauge64
+	partitionLag     map[int32]*stats.Gauge64
+	LagMonitor       *LagMonitor
+	stopChan         chan struct{}
+}
+
+type ConsumerConf struct {
+	ClientID             string
+	Broker               string
+	Partitions           string
+	StartAtOffset        string
+	GaugePrefix          string
+	Topics               []string
+	MessageHandler       func([]byte, int32)
+	BatchNumMessages     int
+	BufferMaxMs          int
+	ChannelBufferSize    int
+	FetchMin             int
+	NetMaxOpenRequests   int
+	MaxWaitMs            int
+	SessionTimeout       int
+	MetadataRetries      int
+	MetadataBackoffTime  int
+	MetadataTimeout      int
+	OffsetCommitInterval time.Duration
+}
+
+func NewConfig() *ConsumerConf {
+	return &ConsumerConf{
+		GaugePrefix:          "default.kafka.partition",
+		BatchNumMessages:     10000,
+		BufferMaxMs:          100,
+		ChannelBufferSize:    1000000,
+		FetchMin:             1,
+		NetMaxOpenRequests:   100,
+		MaxWaitMs:            100,
+		SessionTimeout:       30000,
+		MetadataRetries:      5,
+		MetadataBackoffTime:  500,
+		MetadataTimeout:      10000,
+		OffsetCommitInterval: time.Second * 5,
+	}
+}
+
+func NewConsumer(conf *ConsumerConf) (*Consumer, error) {
+	if len(conf.Topics) < 1 {
+		return nil, fmt.Errorf("kafka-consumer: Requiring at least 1 topic")
+	}
+
+	consumer, err := confluent.NewConsumer(&confluent.ConfigMap{
```

Comment:
would it make sense to use kafka.GetConfig here? why (not) ?

Reply:
:+1: done